pkg/tenant/apikeys.go

```diff
@@ -168,14 +166,78 @@ func (s *Server) APIKeyCreate(c *gin.Context) {
 	c.JSON(http.StatusNotImplemented, "not implemented yet")
 }
 
+// APIKeyDetail returns details about a specific API key.
+//
+// Route: GET /v1/apikeys/:apiKeyID
 func (s *Server) APIKeyDetail(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+	var (
+		ctx context.Context
+		err error
+	)
+
+	// User credentials are required to make the Quarterdeck request
+	if ctx, err = middleware.ContextFromRequest(c); err != nil {
+		log.Error().Err(err).Msg("could not create user context from request")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch credentials for authenticated user"))
+		return
+	}
+
+	// Parse the API key ID from the URL
+	apiKeyID := c.Param("apiKeyID")
+
+	// Get the API key from Quarterdeck
+	// TODO: Handle error status codes returned by Quarterdeck
+	var key *qd.APIKey
+	if key, err = s.quarterdeck.APIKeyDetail(ctx, apiKeyID); err != nil {
+		log.Error().Err(err).Str("apiKeyID", apiKeyID).Msg("could not get API key")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not retrieve API key"))
+		return
+	}
+
+	// Return everything but the client secret
+	out := &api.APIKey{
+		ID:          key.ID.String(),
+		ClientID:    key.ClientID,
+		Name:        key.Name,
+		Owner:       key.CreatedBy.String(),
+		Permissions: key.Permissions,
+		Created:     key.Created.Format(time.RFC3339Nano),
+		Modified:    key.Modified.Format(time.RFC3339Nano),
+	}
+
+	c.JSON(http.StatusOK, out)
 }
 
 func (s *Server) APIKeyUpdate(c *gin.Context) {
 	c.JSON(http.StatusNotImplemented, "not implemented yet")
 }
 
+// APIKeyDelete deletes an API key by forwarding the request to Quarterdeck.
+//
+// Route: DELETE /v1/apikeys/:apiKeyID
 func (s *Server) APIKeyDelete(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+	var (
+		ctx context.Context
+		err error
+	)
+
+	// User credentials are required to make the Quarterdeck request
+	if ctx, err = middleware.ContextFromRequest(c); err != nil {
+		log.Error().Err(err).Msg("could not create user context from request")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch credentials for authenticated user"))
+		return
+	}
+
+	// Parse the API key ID from the URL
+	apiKeyID := c.Param("apiKeyID")
+
+	// Delete the API key using Quarterdeck
+	// TODO: Handle error status codes returned by Quarterdeck
+	if err = s.quarterdeck.APIKeyDelete(ctx, apiKeyID); err != nil {
+		log.Error().Err(err).Str("apiKeyID", apiKeyID).Msg("could not delete API key")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not delete API key"))
+		return
+	}
+
+	c.JSON(http.StatusNoContent, nil)
```

Comment:
Could this be replaced with `c.Status(http.StatusNoContent)` or does nil need to be returned here?

Reply:
Yes, thanks for catching this!